GANDLF/utils/imaging.py

```diff
@@ -1,67 +1,79 @@
-from typing import Union
+from typing import Optional, Union
 import os, pathlib, math, copy
+from enum import Enum
 import numpy as np
 import SimpleITK as sitk
 import torchio
+import cv2
 
 from .generic import get_filename_extension_sanitized
 
 
 def resample_image(
-    img, spacing, size=None, interpolator=sitk.sitkLinear, outsideValue=0
-):
+    input_image: sitk.Image,
+    spacing: Union[np.ndarray, list, tuple],
+    size: Optional[Union[np.ndarray, list, tuple]] = None,
+    interpolator: Optional[Enum] = sitk.sitkLinear,
+    outsideValue: Optional[int] = 0,
+) -> sitk.Image:
     """
-    Resample image to certain spacing and size.
+    This function resamples the input image based on the spacing and size.
+
     Args:
-        img (SimpleITK.Image): The input image to resample.
-        spacing (list): List of length 3 indicating the voxel spacing as [x, y, z].
-        size (list, optional): List of length 3 indicating the number of voxels per dim [x, y, z], which will use compute the appropriate size based on the spacing. Defaults to [].
-        interpolator (SimpleITK.InterpolatorEnum, optional): The interpolation type to use. Defaults to SimpleITK.sitkLinear.
-        origin (list, optional): The location in physical space representing the [0,0,0] voxel in the input image.  Defaults to [0,0,0].
-        outsideValue (int, optional): value used to pad are outside image.  Defaults to 0.
-    Raises:
-        Exception: Spacing/resolution mismatch.
-        Exception: Size mismatch.
+        input_image (sitk.Image): The input image to be resampled.
+        spacing (Union[np.ndarray, list, tuple]): The desired spacing for the resampled image.
+        size (Optional[Union[np.ndarray, list, tuple]], optional): The desired size for the resampled image. Defaults to None.
```

Comment:
same as above with List and Tuple typing

Reply:
Fixed.